dandi/cli/cmd_service_scripts.py

```diff
@@ -0,0 +1,103 @@
+    help="When to re-extract an asset's metadata",
+    show_default=True,
+)
+@click.argument("url")
+@map_to_click_exceptions
+def reextract_metadata(url: str, diff: bool, when: str) -> None:
+    """
+    Recompute & update the metadata for NWB assets on a remote server.
+
+    Running this command requires the fsspec library to be installed with the
+    `http` extra (e.g., `pip install "fsspec[http]"`).
+    """
+    from ..metadata import nwb2asset  # Avoid heavy import at top level
+
+    parsed_url = parse_dandi_url(url)
+    if parsed_url.version_id not in (None, "draft"):
+        raise click.UsageError("URL cannot point to a published Dandiset")
+    parsed_url.version_id = "draft"
+    current_schema_version = Version(DANDI_SCHEMA_VERSION)
+    with parsed_url.navigate(authenticate=True, strict=True) as (_, _, assets):
+        for asset in assets:
+            if PurePosixPath(asset.path).suffix.lower() != ".nwb":
+                lgr.info(
+                    "Asset %s (%s) is not NWB; skipping", asset.identifier, asset.path
+                )
+            assert isinstance(asset, RemoteBlobAsset)
+            lgr.info("Processing asset %s (%s)", asset.identifier, asset.path)
+            if when == "always":
+                do_reextract = True
+            else:
+                try:
+                    sv = asset.get_raw_metadata()["schemaVersion"]
+                except KeyError:
+                    do_reextract = True
+                else:
+                    schemaVersion = Version(sv)
+                    if schemaVersion < current_schema_version:
+                        do_reextract = True
+                    elif schemaVersion == current_schema_version:
+                        lgr.info(
+                            "Asset's schemaVersion is up to date; not reextracting"
+                        )
+                        do_reextract = False
+                    else:
+                        lgr.warning(
+                            "schemaVersion of asset %s (%s) is %r, higher than"
+                            " current schema version %r",
+                            asset.identifier,
+                            asset.path,
+                            sv,
+                            DANDI_SCHEMA_VERSION,
+                        )
+                        do_reextract = False
+            if do_reextract:
+                try:
+                    digest = asset.get_digest()
+                except NotFoundError:
+                    digest = None
+                lgr.info("Extracting new metadata for asset")
+                metadata = nwb2asset(asset.as_readable(), digest=digest)
+                metadata.path = asset.path
+                mddict = metadata.json_dict()
+                if diff:
+                    oldmd = asset.get_raw_metadata()
+                    oldmd_str = json.dumps(oldmd, indent=4, sort_keys=True)
+                    mddict_str = json.dumps(mddict, indent=4, sort_keys=True)
```

Comment:
let's do yaml instead of json here since yaml diff IMHO would be more human readable.